app/lib/actions/candlepin/environment/set_content.rb

```diff
@@ -19,18 +19,32 @@ class SetContent < Candlepin::Abstract
           param :content_ids, Array
         end
 
-        def run
-          saved_cp_ids = ::Katello::Resources::Candlepin::Environment.
+        def existing_ids
+          ::Katello::Resources::Candlepin::Environment.
               find(input[:cp_environment_id])[:environmentContent].map do |content|
             content[:contentId]
           end
+        end
+
+        def run
+          saved_cp_ids = existing_ids
           add_ids    = input[:content_ids] - saved_cp_ids
           delete_ids = saved_cp_ids - input[:content_ids]
-
-          output[:add_ids]         =    add_ids
-          unless add_ids.empty?
-            output[:add_response]    = ::Katello::Resources::Candlepin::Environment.
-              add_content(input[:cp_environment_id], add_ids)
+          retries = 2
+          (retries + 1).times do
```

Comment:
Also, the suggested code handles better the reraising the error when exceeding the number of retries